### PR TITLE
add gb_trees:iterator_from

### PR DESCRIPTION
**Pull request by evilbluebeaver**
gb_trees iterator version useful for traversing from specified Key (all keys greater than specified) without the need to skip previous keys
I coludn't locate any tests for gb_trees, if you direct me I write some tests.

### CODE REVIEW

**Comment by OTP-Maintainer** [comment]
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

**Comment by nox** [comment]
@evilbluebeaver The documentation is in `lib/stdlib/doc/src/gb_trees.xml`, the tests in `lib/stdlib/test/dict_SUITE.erl`.

**Comment by OTP-Maintainer** [comment]
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

**Comment by OTP-Maintainer** [comment]
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

File: lib/stdlib/doc/src/gb_trees.xml [hunk 1]
```diff
@@ -183,6 +183,17 @@
       </desc>
     </func>
     <func>
+      <name name="iterator_from" arity="2"/>
+      <fsummary>Return an iterator for a tree starting from specified key</fsummary>
+      <desc>
+        <p>Returns an iterator that can be used for traversing the
+            entries of <c><anno>Tree</anno></c>; see <c>next/1</c>. The main
+            difference with regular iterator that it finds firts key greater
```
Comment (richcarl) [inline lib/stdlib/doc/src/gb_trees.xml]: "first"

**Comment by richcarl** [comment]
Seems useful. Language in comments/docs needs revising.

**Comment by OTP-Maintainer** [comment]
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

**Comment by uabboli** [comment]
iterator_from() seems like a good thing to have. Thank you!

I've uploaded a branch with some minor modifications, but also with more tests and a new function gb_sets:iterator_from(). See https://github.com/uabboli/otp/tree/evilbluebeaver/iterator_from

It has not yet been tested in our daily builds, but I think it will be merged shortly.

Best regards,

Hans Bolinder, Erlang/OTP team, Ericsson

**Pull request closed without merging**